Formula/libcython.rb

```diff
@@ -0,0 +1,48 @@
+class Libcython < Formula
+  desc "Compiler for writing C extensions for the Python language"
+  homepage "https://cython.org/"
+  url "https://files.pythonhosted.org/packages/cb/da/54a5d7a7d9afc90036d21f4b58229058270cc14b4c81a86d9b2c77fd072e/Cython-0.29.28.tar.gz"
+  sha256 "d6fac2342802c30e51426828fe084ff4deb1b3387367cf98976bb2e64b6f8e45"
+  license "Apache-2.0"
+
+  keg_only <<~EOS
+    this formula is mainly used internally by other formulae.
+    Users are advised to use `pip` to install cython
+  EOS
+
+  depends_on "python@3.10" => [:build, :test]
+  depends_on "python@3.9" => [:build, :test]
+
+  def pythons
+    deps.map(&:to_formula)
+        .select { |f| f.name.match?(/python@\d\.\d+/) }
+        .map(&:opt_bin)
+        .map { |bin| bin/"python3" }
+  end
+
+  def install
+    pythons.each do |python|
+      ENV.prepend_create_path "PYTHONPATH", libexec/Language::Python.site_packages(python)
```

Comment:
For the second (and subsequent) python `PYTHONPATH`  includes site packages from previous pythons. Should we remove it?

Reply:
I think I got weird errors mixing Python versions, and anyway I want to specifically install `cython` libraries twice in this case, once for each Python version. So resetting the `PYTHONPATH` each time is the thing to do.

Reply:
Ah sorry, probably I worded the question badly (or I didn't get the answer).
I wanted to point out that after building cython for python@3.10 we still have `lib/python3.10/site-packages` in PYTHONPATH. And then we build it for python@3.9 — we're adding `lib/python3.9/site-packages` on top of `lib/python3.10/site-packages` in PYTHONPATH, aren't we?

Reply:
Ha yes right got it. `ENV.prepend_create_path` twice is wrong, you are right. I'll update the PR/test this more later today, this needs to be changed.

Reply:
I checked and this is what we have for the first and second loop:
```
/usr/local/Cellar/libcython/0.29.28/libexec/lib/python3.10/site-packages
/usr/local/Cellar/libcython/0.29.28/libexec/lib/python3.9/site-packages:/usr/local/Cellar/libcython/0.29.28/libexec/lib/python3.10/site-packages
```

As we prepend and the first element in the path is used, all is good. Ideally we could add one step to delete the `PYTHONPATH` at the end of the loop but it does not feel necessary.

So I think the code is doing the right thing.

Reply:
Does `PYTHONPATH` do anything in this case? It is usually used to find other packages, but I don't think it matters when the path is for the package that is currently being installed.

`cython` needs it to create env scripts so executables can find unlinked libraries.